src/select.js

```diff
@@ -1,13 +1,13 @@
 import {html} from "htl";
 import {createChooser} from "./chooser.js";
 import {maybeWidth} from "./css.js";
-import {stringify} from "./format.js";
+import {formatAuto} from "./format.js";
 import {maybeLabel} from "./label.js";
 
 export const select = createChooser({
   render(data, index, selected, disabled, {format, multiple, size, label, width}) {
     const select = html`<select class=__ns__-input disabled=${disabled === true} multiple=${multiple} size=${size} name=input>
-      ${index.map(i => html`<option value=${i} disabled=${typeof disabled === "function" ? disabled(i) : false} selected=${selected(i)}>${stringify(format(data[i], i, data))}`)}
+      ${index.map(i => html`<option value=${i} disabled=${typeof disabled === "function" ? disabled(i) : false} selected=${selected(i)}>${formatAuto(format(data[i], i, data))}`)}
```

Comment:
TODO Pass through the locale here, so we can use formatLocaleAuto.

Reply:
Wait, this value is already formatted when we’re formatting it (formatAuto(format(…))), which looks like a mistake.

Reply:
Ah… no, I guess that makes sense, because the user format could be a “bad” format that does not return a string, and so we essentially want to compose it with our auto format. Okay, this is fine.

Reply:
Another wrinkle is that we’re using formatAuto here implicitly, but we don’t in other similar situations such as here in checkbox:

https://github.com/observablehq/inputs/blob/e068507071141de2a5650964c11e381a879d7483/src/checkbox.js#L11

That’s because the *format* you pass to checkbox can return an HTML element, and obviously we don’t want to coerce that to a string. But for select the value returned by format can only be represented as a string. So if we formatAuto here (or formatLocaleAuto), you’re getting a little extra magic here that you wouldn’t get with a checkbox. Which makes me think that maybe we need something like formatAuto, but that passes through `Node` instances without coercing.

Reply:
Ugh, and then if we do that, if we want to be internally consistent, _every_ place we interpolate a user-provided value into an HTML template literal we’ll need to remember to add our implicit auto format. That feels untenable so I need to think through whether the “always” approach in this PR is practical for us to maintain, or if there is some more selective behavior that we should attempt to implement instead. That’s kinda what I suggested on my first response, is that we could address this solely in Inputs.table rather than trying to do it universally.

Reply:
To give a little example of the above problem:

https://github.com/observablehq/inputs/blob/e068507071141de2a5650964c11e381a879d7483/src/label.js#L6

If we want this behavior to be universal, then if your label is a Date instance, then we’d autoformat that here, rather than relying on the implicit toString behavior that HTL provides.